fee_calculator/apps/api/serializers.py

```diff
@@ -8,6 +8,70 @@
 )
 
 
+class SchemeListQuerySerializer(serializers.Serializer):
+    type = serializers.ChoiceField(help_text="Graduated fee scheme type",
+                                   choices=('AGFS', 'LGFS'),
+                                   required=False)
+    case_date = serializers.CharField(help_text="Date for which you would like a list of applicable fee schemes",
+                                      required=False)
+
+
+class BasePriceFilteredQuerySerializer(serializers.Serializer):
+    scenario = serializers.IntegerField(help_text='',
+                                        required=False,)
+    advocate_type = serializers.CharField(help_text='Note the query will return prices with `advocate_type_id` either matching the value or null.',
+                                          required=False,)
+    offence_class = serializers.CharField(help_text='Note the query will return prices with `offence_class_id` either matching the value or null.',
+                                          required=False,)
+    fee_type_code = serializers.CharField(help_text='',
+                                          required=False,)
+
+
+class CalculatorQuerySerializer(serializers.Serializer):
+    class UnitField(serializers.DecimalField):
+        def __init__(self, name):
```

Comment:
Non-blocking comment:
You could accept `**kwargs` and pass them on to the super call.
But might not be as simple as just adding it in both places as if you pass eg `required` in it would try to pass it to the super call twice and this would, I think, cause it to break. True solution is probably accept `**kwargs`, then do `kwargs.update()` with your four overrides, then pass this dict into the super call.
Whether that's worth doing in this case I leave entirely up to you. I mention it mostly as if you were to be using this field in other places you might want to be able to pass the normal keyword args to it like any other field, but you're not, so it might not matter.

Reply:
Oh I was thinking you'd just be doing `kwargs.update(required=False, max_digits=100)` etc which would avoid duplicate keys but yeah that'd stop you being able to pass any of these four kwargs (but you could pass other things not there and it work).

I think there's something less verbose than what you have, and I want to say it's something like this:
```
def __init__(self, name, **kwargs):
    new_kwargs = {'help_text': f'Quantity of the price unit: {name}', 'required': False, 'max_digits': 100, 'decimal_places': 5}
    new_kwargs.update(kwargs)
    super().__init__(**new_kwargs)
```